pkg/accounts/kubernetes/validator.go

```diff
@@ -32,21 +36,45 @@ type kubernetesAccountValidator struct {
 }
 
 func (k *kubernetesAccountValidator) Validate(spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client, ctx context.Context, log logr.Logger) error {
-	if err := k.validateSettings(ctx, log); err != nil {
+	err := k.validateSettings(ctx, log)
+	if err != nil {
 		return err
 	}
-	if k.account.Auth != nil && k.account.Auth.UseServiceAccount && spinSvc == nil {
-		// don't validate if the spinnaker account needs a k8s service account to run validations, and there's no SpinnakerService yet
-		return nil
+	if k.account.Auth != nil && k.account.Auth.UseServiceAccount {
+		spinSvc, err = k.ensureSpinSvc(spinSvc, c)
+		if err != nil {
+			return err
+		}
+		if spinSvc == nil {
+			// don't validate if the spinnaker account needs a k8s service account to run validations, and there's no SpinnakerService yet
+			return nil
+		}
 	}
-	config, err := k.makeClient(ctx, spinSvc)
+	config, err := k.makeClient(ctx, spinSvc, c)
 	if err != nil {
 		return err
 	}
 	return k.validateAccess(config)
 }
 
-func (k *kubernetesAccountValidator) makeClient(ctx context.Context, spinSvc v1alpha2.SpinnakerServiceInterface) (clientcmd.ClientConfig, error) {
+func (k *kubernetesAccountValidator) ensureSpinSvc(spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client) (v1alpha2.SpinnakerServiceInterface, error) {
+	if spinSvc != nil {
+		return spinSvc, nil
+	}
+	i := SpinnakerServiceBuilder.NewList()
+	list, err := util.GetSpinnakerServices(i, k.account.CrdNamespace, c)
```

Comment:
use `spinSvc.getNamespace()`. An account must be in the same namespace as the spinnaker service. `CrdNamespace` is not necessary.